### PR TITLE
alternative model for xgboost

### PR DESCRIPTION
**Pull request by tolothy**
replace xgboost.XGBRegressor by sklearn.ensemble.GradientBoostingRegressor when xgboost is not an feasible option (for Windows users)

Performance is comparable.

### CODE REVIEW

File: api/client/samples/prevented_plant/prevented_plant_xgboost.ipynb [hunk 1]
```diff
@@ -1761,8 +1763,13 @@
     "Out of sample testing (leave one year out)\n",
     "\"\"\"\n",
     "from xgboost import XGBRegressor\n",
+    "from sklearn.ensemble import GradientBoostingRegressor\n",
     "from sklearn.metrics import r2_score, mean_squared_error, mean_absolute_error\n",
     "\n",
+    "gbt_model = XGBRegressor()\n",
+    "# if the above line does not work, use the following line instead\n",
```
Comment (muzigao) [inline api/client/samples/prevented_plant/prevented_plant_xgboost.ipynb]: is there any way to automatically detect XGBRegressor() work or not? ?

**Review by muzigao — APPROVED** [review]
lgtm

**Comment by tolothy** [comment]
Updated the PR with two more changes:
1. move the importance and construction of ensemble boosting models to initial set up, and add instructions.
2. move the importance of geopandas to initial set up, and add instructions.

File: api/client/samples/prevented_plant/prevented_plant_xgboost.ipynb [hunk 2]
```diff
@@ -34,6 +34,60 @@
    "source": [
```
Comment (JGrinst) [inline api/client/samples/prevented_plant/prevented_plant_xgboost.ipynb:34]: <p>"For Windows users, geopandas is best installed by using anaconda and the command `conda install geopandas'. Please see <a href="http://geopandas.org/install.html" target="_blank">http://geopandas.org/install.html</a> for further details.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gro-intelligence/api-client/pull/86/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (iba3) [reply api/client/samples/prevented_plant/prevented_plant_xgboost.ipynb:34]: Does conda install also install shapely?

Comment (tolothy) [reply api/client/samples/prevented_plant/prevented_plant_xgboost.ipynb:34]: updated instruction

**Review by iba3 — APPROVED** [review]

**Review by JGrinst — APPROVED** [review]

**Pull request merged**